website/js/app/controllers/content.js

```diff
@@ -0,0 +1,182 @@
+  // Child controllers can subscribe to the message as needed.
+  $scope.onParameterSetSelected = function(parameterSetName) {
+    $scope.$broadcast("parameterSetSelected", parameterSetName);
+  };
+  // onParameterSetDeselected works the same, but broadcases out a 'null' parameter set.
+  $scope.onParameterSetDeselected = function() {
+    $scope.$broadcast("parameterSetSelected", null);
+  };
+
+  // There are some quirks with the way our script generates JSON data. Specifically, arrays
+  // with a single element are serialized as the object and not an array with one object.
+  // We manually fix these objects up so that we avoid needing to special case the controllers.
+  //
+  // The function takes a string similar to an object query, to walk down an object hierarchy.
+  // * meaning all fields.
+  this._makeArray = function(path, sourceObj) {
+    // Convert the field into an array if needed.
+    function makeFieldArray(fieldName, obj) {
+      var fieldValue = obj[fieldName];
+      if (fieldValue == null) {
+        obj[fieldName] = [];
+      } else if (!Array.isArray(fieldValue)) {
+        obj[fieldName] = [fieldValue];
+      }
+    }
+
+    // Walk down the object properties until you get to the specific field you want to upgrade.
+    // @type {Array.<string>} fields The array of fields to check.
+    // @type {Object} obj The object to update.
+    function walkTypeProperties(fields, obj) {
+      if (!obj) {
+        return;
+      }
+      // If the object is an array, walk down all ements of it.
+      if (Array.isArray(obj)) {
+        for (var i = 0; i < obj.length; i++) {
+          walkTypeProperties(fields, obj[i]);
+        }
+        return;
+      }
+
+      var nextFieldName = fields[0];
+      // Done walking the type. Finally make the field an array.
+      if (fields.length == 1) {
+        // BUG: If the fields list ends in a '*', we don't do the right thing.
+        makeFieldArray(nextFieldName, obj)
+        return;
+      }
+      // Recurse walking down the next field. The field name '*' means to recurse checking
+      // all object properties.
+      if (nextFieldName != '*') {
+        walkTypeProperties(fields.slice(1), obj[nextFieldName]);
+      } else {
+        for (var prop in obj) {
+          if (obj.hasOwnProperty(prop)) {
+            walkTypeProperties(fields.slice(1), obj[prop]);
+          }
+        }
+      }
+    }
+
+    var parts = path.split('.');
+    walkTypeProperties(parts, sourceObj);
+  };
+
+  // Populate member variables with the right data based on routeParams and cmdletDocumentation.
+  // Call this after this.cmdletDocumentation has been set.
+  this._loadContent = function() {
+    if (!this.cmdletDocumentation) return;
+
+    // Hierarchy of resources from Home > Product > Resource > Cmdlet.
+    this.contentHeader =
+        this.currentCmdlet || this.currentResource ||
+        this.currentProduct || "Google Cloud Tools for PowerShell";
```

Comment:
There are a fair amount of strings just laying around.  Maybe make them all constants?